From 45dd6be0c65769b6eec5830c61662000ed389aad Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Sun, 23 Mar 2025 11:03:46 +0800
Subject: [PATCH 4/7] configurations: nvidia_gb200_io_board: Use external
 sensor service for temperature monitoring

Due to a hardware design limitation, the port temperature is only
accessible via pldmtool.

Switch Backend NIC die temperature and port temperature monitoring to
external sensor service, this allows a other service to update sensor
readings.

Signed-off-by: Potin Lai <potin.lai@quantatw.com>
Change-Id: Iee90f1b75cf3a7ddabcd5f5cd640897431d4a6ea
---
 .../nvidia_cx7_mezzanine_module.json          | 90 ++++++++++++++-----
 1 file changed, 70 insertions(+), 20 deletions(-)

diff --git a/configurations/nvidia_cx7_mezzanine_module.json b/configurations/nvidia_cx7_mezzanine_module.json
index 7a00978..1a07e06 100644
--- a/configurations/nvidia_cx7_mezzanine_module.json
+++ b/configurations/nvidia_cx7_mezzanine_module.json
@@ -7,72 +7,122 @@
             "Type": "EEPROM"
         },
         {
-            "Address": "0x1f",
-            "Bus": "$bus - 5",
-            "Labels": [
-                "temp2"
-            ],
             "Name": "IOB$bus - 29 % 11 NIC0_TEMP_C",
-            "Name1": "IOB$bus - 29 % 11 NIC0_TEMP_C",
+            "Units": "xyz.openbmc_project.Sensor.Value.Unit.DegreesC",
+            "MinValue": -255.0,
+            "MaxValue": 255,
+            "Timeout": 30.0,
             "PowerState": "On",
             "Thresholds": [
                 {
                     "Direction": "greater than",
-                    "Index": 2,
+                    "Hysteresis": 2.16,
                     "Name": "upper critical",
                     "Severity": 1,
                     "Value": 108.0
                 },
                 {
                     "Direction": "greater than",
-                    "Index": 2,
+                    "Hysteresis": 2.1,
                     "Name": "upper non critical",
                     "Severity": 0,
                     "Value": 105.0
                 },
                 {
                     "Direction": "less than",
-                    "Index": 2,
+                    "Hysteresis": 0.1,
                     "Name": "lower non critical",
                     "Severity": 0,
                     "Value": 5
                 }
             ],
-            "Type": "TMP421"
+            "Type": "ExternalSensor"
         },
         {
-            "Address": "0x1f",
-            "Bus": "$bus - 3",
-            "Labels": [
-                "temp2"
-            ],
             "Name": "IOB$bus - 29 % 11 NIC1_TEMP_C",
-            "Name1": "IOB$bus - 29 % 11 NIC1_TEMP_C",
+            "Units": "xyz.openbmc_project.Sensor.Value.Unit.DegreesC",
+            "MinValue": -255.0,
+            "MaxValue": 255,
+            "Timeout": 30.0,
             "PowerState": "On",
             "Thresholds": [
                 {
                     "Direction": "greater than",
-                    "Index": 2,
+                    "Hysteresis": 2.16,
                     "Name": "upper critical",
                     "Severity": 1,
                     "Value": 108.0
                 },
                 {
                     "Direction": "greater than",
-                    "Index": 2,
+                    "Hysteresis": 2.1,
                     "Name": "upper non critical",
                     "Severity": 0,
                     "Value": 105.0
                 },
                 {
                     "Direction": "less than",
-                    "Index": 2,
+                    "Hysteresis": 0.1,
                     "Name": "lower non critical",
                     "Severity": 0,
                     "Value": 5
                 }
             ],
-            "Type": "TMP421"
+            "Type": "ExternalSensor"
+        },
+        {
+            "Name": "IOB$bus - 29 % 11 NIC0_OSFP_TEMP_C",
+            "Units": "xyz.openbmc_project.Sensor.Value.Unit.DegreesC",
+            "MinValue": -255.0,
+            "MaxValue": 255,
+            "Timeout": 30.0,
+            "PowerState": "On",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.4,
+                    "Index": 2,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 70.0
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.1,
+                    "Index": 2,
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 5
+                }
+            ],
+            "Type": "ExternalSensor"
+        },
+        {
+            "Name": "IOB$bus - 29 % 11 NIC1_OSFP_TEMP_C",
+            "Units": "xyz.openbmc_project.Sensor.Value.Unit.DegreesC",
+            "MinValue": -255.0,
+            "MaxValue": 255,
+            "Timeout": 30.0,
+            "PowerState": "On",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.4,
+                    "Index": 2,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 70.0
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.1,
+                    "Index": 2,
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 5
+                }
+            ],
+            "Type": "ExternalSensor"
         }
     ],
     "Name": "NVIDIA GB200 IO Board $bus - 29 % 11",
-- 
2.47.3

